src/script/sign.cpp

```diff
@@ -81,30 +59,40 @@ static bool SignStep(const SigningProvider& provider, const BaseSignatureCreator
     case TX_WITNESS_UNKNOWN:
         return false;
     case TX_PUBKEY:
-        keyID = CPubKey(vSolutions[0]).GetID();
-        return Sign1(provider, keyID, creator, scriptPubKey, ret, sigversion);
-    case TX_PUBKEYHASH:
+        if (!creator.CreateSig(provider, sig, CPubKey(vSolutions[0]).GetID(), scriptPubKey, sigversion)) return false;
+        ret.push_back(std::move(sig));
+        return true;
+    case TX_PUBKEYHASH: {
         keyID = CKeyID(uint160(vSolutions[0]));
```

Comment:
nit: This is the only use of `keyID` now, so you could declare it in this block.

Reply:
Done